compiler/src/dmd/expressionsem.d

```diff
@@ -1155,6 +1155,43 @@ L1:
     return e1;
 }
 
+/*
+ * Check whether `outerFunc` and `calledFunc` have the same `this`.
+ * If `calledFunc` is the member of a base class of the class that contains
+ * `outerFunc` we consider that they have the same this.
```

Comment:
The example makes it unclear what "have same `this`" means, it sounds more like "have covariant `typeof(this)`"

Reply:
Well, I followed the existing naming pattern after `getRightThis`.